### ISSUE TITLE
GET with passport.authenticate & JWT Authorization header not being called

### ISSUE THREAD

**Issue by ttbarnes** [themikenicholson/passport-jwt#89]
I've updated some dependencies and now my protected GET endpoints are no longer working :(

Does anyone know why this could be happening, and how to resolve?

EG: 

```
router.route('/private')
  .get(
    passport.authenticate('jwt', { session: false }), (req, res, next) => {
      res.json({ allTheThings: true });
  })
```

If I remove the `passport.authenticate` and simply send a json response, it works as expected.

With `passport.authenticate` included - if I send a JWT Authorization header, I get nothing back - nothing at all. In unit tests this can also result in 'socket hang up' errors.

If I remove the JWT Authorization header, still with `passport.authenticate`, it returns unauthorized, as expected.

Any help appreciated!

Relevant dependancies: 

```
"body-parser": "^1.14.2",
"express": "^4.14.0",
"passport": "^0.3.2",
"passport-jwt": "^2.2.1",
```

This is happening with node 6.9.2 and 7.

**Comment by themikenicholson** [comment]
Is this still an issue? I see that it was closed and reopened. 

Without seeing your code or an example program that reproduces the issue there isn't enough information here for me to help you.

There could be a number of reasons why your jwt is not validating. Could it be expired? Jwt's are generally created with an expiration time.  If you're creating it and storing it in the DB(you mention creating it with a user schema??) then perhaps it is expiring after some period and that is why it no longer works. 

Again, I'm just guessing. There is not nearly enough information here.

**Comment by ttbarnes** [comment]
Thanks @themikenicholson - this is no longer an issue, I reopened just incase someone had some ideas about what could cause this.

We added expiration to the JWT, however In my test environment, the user would have got a fresh JWT created, before other auth-only endpoints would be hit.

There was some missing fields in the user. I guess that somehow between adding new fields to the user schema and adding a JWT expiry, the specific User/JWT in question would not work. Maybe something to do with the encryption side.

If I come across this issue again i'll create an example.

**Comment by fromage9747** [comment]
@ttbarnes I just encountered this issue with the project I am working on.  It was with a user account that I created when I first built the app however the JWT expiry has been the same since the start.  Not sure what caused it.  I too would be interested in how it happened.  Spent two days with troubleshooting ruling out other possibilities.  I was able to successfully login with a newer user account.
